assets/src/components/ladderPage.tsx

```diff
@@ -24,9 +38,53 @@ export const findSelectedVehicleOrGhost = (
   )
 }
 
-const LadderPage = (): ReactElement<HTMLDivElement> => {
-  const [state] = useContext(StateDispatchContext)
-  const { selectedRouteIds, selectedVehicleOrGhost } = state
+const LadderTab = ({
+  tab,
+  selectTab,
+}: {
+  tab: RouteTab
+  selectTab: () => void
+}): ReactElement<HTMLDivElement> => {
+  const title = tab.presetName || "Untitled"
+  return (
+    <div
+      className={
+        tab.isCurrentTab ? "m-ladder-page__tab-current" : "m-ladder-page__tab"
+      }
+      onClick={() => selectTab()}
+    >
+      {title}
+    </div>
+  )
+}
+
+const AddLadderButton = ({
+  addLadder,
+}: {
+  addLadder: () => void
+}): ReactElement<HTMLDivElement> => {
+  return (
+    <div className="m-ladder-page__add-tab-button" onClick={addLadder}>
+      +
+    </div>
+  )
+}
+
+const LadderPage = (): ReactElement<HTMLDivElement> =>
+  featureIsEnabled("presets_workspaces") ? (
+    <LadderPageWithTabs />
+  ) : (
+    <LadderPageWithoutTabs />
+  )
+
+const LadderPageWithoutTabs = (): ReactElement<HTMLDivElement> => {
+  const [state, dispatch] = useContext(StateDispatchContext)
```

Comment:
**thought**: another thing about reducers, we have some in the jumbo reducer associated with `StateDispatchContext`, and then some more special-purpose reducers here and there. I wonder if there would be any benefit to consolidating those loose reducers into the big one rather than having them scattered.

Reply:
Right, we've got a few. I kind of like the way that the search reducer works, where it's part of the main state but has its own reducer function that's defined in a separate module etc.